src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private static final int NUMBER_OF_BULLDOZERS = 5;
+    private static final int NUMBER_OF_EXCAVATORS = 6;
+    private static final int NUMBER_OF_TRUCKS = 7;
+    private BulldozerProducer bulldozerProducer = new BulldozerProducer();
+    private TruckProducer truckProducer = new TruckProducer();
+    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = new ArrayList<>();
```

Comment:
`List<? extends Machine>` in local context works in another way. 

Make it `List<Machine>`.

see a link to understand the difference : https://stackoverflow.com/questions/2776975/how-can-i-add-to-list-extends-number-data-structures